### ISSUE TITLE
Auto trigger snippets

### ISSUE THREAD

**Issue by maikol-solis** [rstudio/rstudio#6834]
Rstudio can handle customized snippets for a lot of languages, but they always trigger using TAB or SHIFT-TAB depending on the file (commonly R or Rmd files).

I would like to know if Rstudio could trigger those snippets only by typing the characters of the snippets. 

For example, imaging you have this snippet 

```
snippet eq
	\begin{equation}
		${1}
	\end{equation}
``` 

So it would awesome if only by typing `eq` the snippet triggers automatically. 

I think it should be some rules about the context if some snippet is triggered or not. I think three mainly (could be others?): 
- typing math text, 
- normal text 
- a code chunk.

I'm thinking in something like this video especially for typing documents fast with a lot of math and code mixed.  (He uses vim for typing pure Latex documents, but you get the idea).

https://www.youtube.com/watch?v=a7gpx0h-BuU

Additional info about how he did it is in this link: https://castel.dev/post/lecture-notes-1/

I asked the same question in stackoverflow with no success: https://stackoverflow.com/questions/61616131/is-it-possible-to-auto-trigger-snippets-in-rstudio

Thanks for the help.

**Comment by jmcphers** [comment]
This is a neat idea (and as a Vim nerd myself I really enjoyed the background!) but not likely something we can do soon in RStudio.  The rules governing in which context a snippet should be auto-expanded are complicated, and likely to require some fine tuning by end users (as even Ultisnips needed custom code added in the case above!).

**Comment by maikol-solis** [comment]
Hi @jmcphers. I totally understand the context and I agree that it could be difficult. 

Other softwares have handled this problem using the way it highlight the code. For example, [Texstudio](http://texstudio.sourceforge.net/) uses the highlight internal system to determine in which context some snippet should be triggered. A great explanation is here: https://tex.stackexchange.com/a/158889/215059

I made some of them using this trick and works perfectly. Look for the "trigger" item: https://github.com/maikol-solis/FastLatex-TeXStudio. I am working on making some for Rstudio. 

In my personal case for a Rmd, I opened with TeXstudio to type the math stuff and then I switch to Rstudio to type the coding stuff. Sometimes I have to switch back and forth multiple times.  This workflow would be easier if I could do everything directly in Rstudio.
